backbone/src/main/java/org/researchstack/backbone/ui/ConsentTaskActivity.kt

```diff
@@ -0,0 +1,219 @@
+                        firstName = nameResult.result as String
+                    }
+                }
+                if (question.identifier.equals(ID_FORM_LAST_NAME, true)) {
+                    val nameResult = result?.getResultForIdentifier(ID_FORM_LAST_NAME) as StepResult<*>?
+                    if (nameResult != null && nameResult.result != null) {
+                        lastName = nameResult.result as String
+                    }
+                }
+            }
+        }
+        if (step is ConsentSignatureStep) {
+            signatureBase64 = result?.getResultForIdentifier(KEY_SIGNATURE) as String
+        } else if (step is ConsentDocumentStep) {
+            consentHtml = step.consentHTML
+        }
+    }
+
+    private fun getFormalName(firstName: String?, lastName: String?): String? {
+        return if (lastName != null && firstName != null) {
+            "$lastName, $firstName"
+        } else {
+            firstName ?: lastName
+        }
+    }
+
+    override fun close(completed: Boolean) {
+        // you can also set title / message
+        val dialog = AlertDialog.Builder(this).setCancelable(false)
+                .setTitle(LocaleUtils.getLocalizedString(this, R.string.rsb_saving_consent))
+                .setMessage(LocaleUtils.getLocalizedString(this, R.string.rsb_please_wait)).create()
+
+        dialog.show()
+
+        val consentAssetsFolder = intent.getStringExtra(EXTRA_ASSETS_FOLDER)
+        val role = LocaleUtils.getLocalizedString(this, R.string.rsb_consent_role)
+        val df = DateFormat.getDateInstance(DateFormat.MEDIUM,
+                LocaleUtils.getLocaleFromString(LocaleUtils.getPreferredLocale(this)))
+        consentHtml += getSignatureHtmlContent(getFormalName(firstName, lastName), role, signatureBase64,
+                df.format(Date()))
+
+        PDFWriteExposer().printPdfFile(this, getCurrentTaskId(), consentHtml!!, consentAssetsFolder) {
+            dialog.dismiss()
+            super.close(completed)
+        }
+    }
+
+    private fun getSignatureHtmlContent(completeName: String?, role: String?, signatureB64: String?,
+                                        signatureDate: String?): String {
+        val body = StringBuffer()
+
+        val hr = "<hr align='left' width='100%' style='height:1px; border:none; color:#000; background-color:#000; margin-top: 5px; margin-bottom: 0px;'/>"
+        val td = "<td style='vertical-align: bottom;" + "width: 33%;" + "padding-right: 10px;" + "padding-left: 10px;'>"
+
+        val signatureElementWrapper = "<div class='sigbox'><div class='inbox'>%s</div></div>"
+        val imageTag: String
+
+        val table = ArrayList<Array<String>>()
+
+        if (role == null) {
+            throw RuntimeException("Consent role cannot be empty")
+        }
+
+        // Signature
+        if (completeName != null) {
+            val columnName = getString(R.string.rsb_consent_doc_line_printed_name, role)
```

Comment:
This rsb_consent_doc... line is in error for me. Apparently the format string is not valid. Can you dbl check that please?

Reply:
I've already reported this to Pablo, Keith, and Norberto, the format for `nl-rBE` is incorrect and needs to be re-localized, so this will be included in 4.9.1